tests/test_metadata_serilization.py

```diff
@@ -0,0 +1,201 @@
+# Copyright New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+""" Unit tests testing tuf/api/metadata.py classes
+serialization and deserialization.
+
+"""
+
+import json
+import sys
+import logging
+import unittest
+import copy
+
+from typing import Any, Dict, Callable, Type
+
+from tests import utils
+
+from tuf.api.metadata import (
+    Root,
+    Snapshot,
+    Timestamp,
+    Targets,
+    Key,
+    Role,
+    MetaFile,
+    TargetFile,
+    Delegations,
+    DelegatedRole,
+)
+
+logger = logging.getLogger(__name__)
+
+# DataSet is only here so type hints can be used:
+# It is a dict of name to test dict
+DataSet = Dict[str, Dict[str, Any]]
+
+# Test runner decorator: Runs the test as a set of N SubTests,
+# (where N is number of items in dataset), feeding the actual test
+# function one data item at a time
+def run_sub_tests_with_dataset(dataset: Type[DataSet]):
+    def real_decorator(function: Callable[["TestSerialization", DataSet], None]):
+        def wrapper(test_cls: "TestSerialization"):
+            for case, data in dataset.items():
+                with test_cls.subTest(case=case):
+                    function(test_cls, data)
+        return wrapper
+    return real_decorator
+
+
+class TestSerialization(unittest.TestCase):
+
+    KEY = '{"keytype": "rsa", "scheme": "a", "keyval": {"public": "foo"}}'
+    KEYS = f'"keys": {{"id" : {KEY} }}'
+    HASHES = '"hashes": {"sha256" : "abc"}'
+    SIGNED_COMMON = '"spec_version": "1.0.0", "version": 1, \
+        "expires": "2030-01-01T00:00:00Z"'
+
+    valid_keys = {
```

Comment:
Would it make this more clear if we annotated:
```suggestion
    valid_keys: Dataset = {
```